src/Type/Constant/ConstantArrayType.php

```diff
@@ -573,7 +574,7 @@ private function removeLastElements(int $length): self
 			if ($i >= $newLength) {
 				if ($isOptional) {
 					$optionalKeysRemoved++;
-					unset($optionalKeys[$i]);
+					unset($optionalKeys[array_search($i, $optionalKeys, true)]);
```

Comment:
I really don't like the usage of array_search like that. If it's not going to find anything, it's going to unset index 0.

Just search for the key manually 😊

Reply:
yeah me too, I was already looking at another approach, give me a minute